src/pydata_sphinx_theme/__init__.py

```diff
@@ -131,19 +131,39 @@ def generate_nav_html(kind, startdepth=None, show_nav_level=1, **kwargs):
             for ul in soup("ul", recursive=False):
                 ul.attrs["class"] = ul.attrs.get("class", []) + ["nav", "bd-sidenav"]
 
+            # Add collapse boxes for parts/captions.
+            # show_nav_level: 0 means make parts collapsible.
+            if not show_nav_level:
+                partcaptions = soup.find_all("p", attrs={"class": "caption"})
+                if len(partcaptions):
+                    new_soup = bs("<ul class='list-caption'></ul>", "html.parser")
+                    for caption in partcaptions:
+                        for sibling in caption.next_siblings:
+                            if sibling.name == "ul":
+                                toclist = sibling
+                                break
+                        li = soup.new_tag("li", attrs={"class": "toctree-l0"})
+                        li.append(caption)
+                        li.append(toclist)
+                        new_soup.ul.append(li)
```

Comment:
Instead of re-using `new_soup` everywhere else, why not just add one extra line here of

```python
soup = new_soup
```

and then you can remove the last two `else` blocks here and just re-use `soup` everywhere?

Reply:
I was trying to establish explicitly that the soup will change here, but it does not work for else statements I agree.